#158174587 test accounts

matthewacha opened the pull request:
#### What does this PR do?
- Fixes unit tests in the directory hc/accounts/tests
- Adds new tests for untested components
 standards
#### Description of Task to be completed?
Implementing the following test cases
- test_check_token.py
  - test_redirects_already_logged_in
  - test_login_bad_token_redirects
- test_login.py
  - test_it_redirects_when_password_given
- test_profile.py
 - test_it_sends_set_password_link
   - Assert that the token is set
   - Assert that the email was sent and check email content
 - test_it_sends_report
   - Assert that the email was sent and check email content
 - test_it_adds_team_member
   - Assert the existence of the member emails
   - Assert that the email was sent and check email content
 - test_it_creates_and_revokes_api
 - test_it_updates_reports
-test_switch_team.py
 - test_it_switches
   - Assert the contents of r
 - test_it_checks_team_membership
   - Assert the expected error code
 - test_it_switches_to_own_team
   - Assert the expected response code
 - test_team_access_middleware.py
   - test_it_handles_missing_profile
     -Assert the new Profile objects count

#### How should this be manually tested?
After cloning the repo, run the following commands
- make
- python manage.py test
#### Any background context you want to provide?
This is necessary before any further code implementation is made to the code base
#### What are the relevant pivotal tracker stories?
[#158174587](https://www.pivotaltracker.com/n/projects/2177206)

matthewacha committed:
[Feature #158174587] Add test for redirect already logged in

matthewacha committed:
[Feature #158174587] Add test for redirect already logged in

matthewacha committed:
[Feature #158174587] Add test bad token login redirects

matthewacha committed:
[Feature #158174587] Add test bad token login redirects

matthewacha committed:
[Feature #158174587] Add assert contents of email body

matthewacha committed:
[Feature #158174587] Add assert contents of email body

matthewacha committed:
[Feature #158174587] Add doc strings and refactor tests

matthewacha committed:
[Feature #158174587] Add doc strings and refactor tests

matthewacha committed:
[Feature #158174587] Add asserts and new tests

matthewacha committed:
[Feature #158174587] Add asserts and new tests

matthewacha committed:
[feature] fix accounts test

add asserts in the test cases
add doc strings
breakdown the tests so that each test case tests single element.
add other tests

[Delivers #158174587]

matthewacha committed:
[feature] fix accounts test

add asserts in the test cases
add doc strings
breakdown the tests so that each test case tests single element.
add other tests
aplly reverse for urls

[Delivers #158174587]

matthewacha committed:
[feature] fix accounts test

add asserts in the test cases
add doc strings
breakdown the tests so that each test case tests single element.
add other tests
aplly reverse for urls

[Delivers #158174587]

matthewacha committed:
[feature] fix accounts test

add asserts in the test cases
add doc strings
breakdown the tests so that each test case tests single element.
add other tests

[Delivers #158174587]

magicmarie commented:
Nice one on the docstrings. The rest of us should pick a leaf

magicmarie committed:
#158174586 chore(api-tests) Fix API tests (#1)

* chore(api-tests) Fix API tests

This commit fixes all tests in hc.api.tests

Fix tests for sending alerts, notifications, checking models

[Delivers #158549842]

* chore(heroku-deployment)Deploy the application to heroku
This commit  deploys the application to heroku

[Delivers #158549842]

* This commit fixes the test errors in hc.api.tests from PR comments

Fixes tests for set Alice as staff and superuser, assign all channels,
Renames variables,  check for kind pushbullet in the channel objects

[Delivers #158549842]

* chore(api-tests) Fix API tests

This commit fixes all tests in hc.api.tests

Fix tests for sending alerts, notifications, checking models

[Delivers #158549842]

* This commit fixes the test errors in hc.api.tests from PR comments

Fixes tests for set Alice as staff and superuser, assign all channels,
Renames variables,  check for kind pushbullet in the channel objects

[Delivers #158549842]

matthewacha committed:
[feature] Set-up continuous integration with your preferred tool of choice (#6)

- Add travis build badge to readme
- Add slack notifications from travis after evry failed build
- Add coveralls badge to README

[Delivers #158549881

EdwinKato commented on hc/accounts/tests/test_check_token.py:
```diff
@@ -10,19 +10,30 @@ def setUp(self):
+        self.client.post("/accounts/check_token/alice/secret-token/")
+        re = self.client.post("/accounts/check_token/alice/secret-token/")
+        self.assertEqual(re.status_code, 302)
+        self.assertRedirects(re, "/checks/")
```
Use reverse for these urls `reverse("hc-checks")`

EdwinKato commented on hc/accounts/tests/test_check_token.py:
```diff
@@ -10,19 +10,30 @@ def setUp(self):
+        #Login with a bad token and check that it redirects
+        self.client.post("/accounts/login/", {"email":"bob@example.org"},follow=True)
+        response_bad_token = self.client.post("/accounts/check_token/bob/bad-token/")
+        self.assertRedirects(response_bad_token,"/accounts/login/")
```
You can do without line 37. Leave a space `response_bad_token`

matthewacha committed:
Merge branch 'ft-test-accounts-#158174587' of https://github.com/andela/hc-aces-kla into ft-test-accounts-#158174587

matthewacha committed:
[feature #158174587] Apply reverse url

matthewacha committed:
[feature #158174587] Apply reverse url

matthewacha committed:
Ch setup ci #158549881 (#9)

* [feature] Set-up continuous integration with your preferred tool of choice

- Add travis build badge to readme
- Add slack notifications from travis after evry failed build
- Add coveralls badge to README

[Delivers #158549881

* [feature] Set-up continuous integration with your preferred tool of choice

- Add travis build badge to readme
- Add slack notifications from travis after evry failed build
- Add coveralls badge to README
- Remove python 2

[Delivers #158549881

EdwinKato commented:
@matthewacha resolve conflicts

matthewacha committed:
[feature] Resolve conflicts with develop
- match requirements and ignore to develop branch

[Delivers #158174587]
